main.tf

```
@@ -1,13 +1,99 @@
-resource "random_integer" "example" {
-  count = module.this.enabled ? 1 : 0
-
-  min = 1
-  max = 50000
-  keepers = {
-    example = var.example
+locals {
+  records = try(jsondecode(var.records), var.records)
+  record_objects = {
+    for i, r in local.records :
+    join("_", compact([
+      try(lower(r.name), ""),
+      lower(r.type),
+      try(lower(r.set_identifier), ""),
+      try(lower(r.failover), "")
+      ])) => {
+      name = try(lower(r.name), "")
+      type = r.type
+      ttl  = try(r.ttl, null)
+      records = try(r.records, null)
+      alias = {
+        name                   = try(r.alias.name, null)
+        zone_id                = try(r.alias.zone_id, null)
+        evaluate_target_health = try(r.alias.evaluate_target_health, null)
+      }
+      geolocation = {
+        continent   = try(r.geolocation.continent, null)
+        country     = try(r.geolocation.country, null)
+        subdivision = try(r.geolocation.subdivision, null)
+      }
+      allow_overwrite = try(r.allow_overwrite, var.allow_overwrite)
+      health_check_id = try(r.health_check_id, null)
+      set_identifier  = try(r.set_identifier, null)
+      weight          = try(r.weight, null)
+      failover        = try(r.failover, null)
+    }
   }
 }
 
-locals {
-  example = format("%v %v", var.example, join("", random_integer.example[*].result))
+resource "aws_route53_zone" "default" {
+  name          = var.zone_name
```

Use enabled logic to disable resources

```suggestion
  count = local.enabled ? 1 : 0

  name          = var.zone_name
```